Soccar [UI]/src/nl/soccar/ui/input/GamePad.java

```diff
@@ -0,0 +1,161 @@
+package nl.soccar.ui.input;
+
+import java.util.Timer;
+import java.util.TimerTask;
+import net.java.games.input.Component;
+import net.java.games.input.Controller;
+import nl.soccar.library.enumeration.HandbrakeAction;
+import nl.soccar.library.enumeration.SteerAction;
+import nl.soccar.library.enumeration.ThrottleAction;
+
+/**
+ *
+ * @author PTS34A
+ */
+public final class GamePad {
+
+    private static final int ACTION_STEER_LEFT = 40;
+    private static final int ACRION_STEER_RIGHT = 60;
+    private static final int ACTION_THROTTLE = 45;
+    private static final int ACTION_REVERSE = 55;
+    private static final int POLL_TIME = 50;
+
+    private Controller controller;
+
+    private ThrottleAction throttleAction;
+    private HandbrakeAction handbrakeAction;
+    private SteerAction steerAction;
+
+    private boolean POLL = false;
+    private Timer timer = new Timer(true);
+    private int xAxisPercentage;
+
+    public GamePad() {
+        throttleAction = ThrottleAction.IDLE;
+        handbrakeAction = HandbrakeAction.INACTIVE;
+        steerAction = SteerAction.NONE;
+    }
+
+    public GamePad(Controller controller) {
+        this.controller = controller;
+        POLL = true;
+
+        timer.scheduleAtFixedRate(new TimerTask() {
+            @Override
+            public void run() {
+                polling(controller);
+            }
+        }, 0, POLL_TIME);
+    }
+
+    private void polling(Controller controller) {
+        if (!POLL) {
+            return;
+        }
+        if (!controller.poll()) {
+            POLL = false;
+            return;
+        }
+
+        Component[] components = controller.getComponents();
+        for (Component c : components) {

```

Comment:
Kan in principe ook `for (Component c : controller.getComponents())` zijn omdat components maar één keer wordt gebruikt.